pytorch_lightning/loops/dataloader/evaluation_loop.py

```diff
@@ -219,10 +220,17 @@ def _get_max_batches(self) -> List[int]:
 
     def _reload_evaluation_dataloaders(self) -> None:
         """Reloads dataloaders if necessary."""
+        dataloaders = None
+
         if self.trainer.testing:
             self.trainer.reset_test_dataloader()
+            dataloaders = self.trainer.test_dataloaders
         elif self.trainer.val_dataloaders is None or self.trainer._data_connector._should_reload_val_dl:
             self.trainer.reset_val_dataloader()
+            dataloaders = self.trainer.val_dataloaders
+
+        if dataloaders:
```

Comment:
```suggestion
        if dataloaders is not None:
```

just a nit: but let's be precise :)

Reply:
did this way to make sure `[]` are handled too